zerver/lib/markdown/api_arguments_table_generator.py

```diff
@@ -194,3 +209,33 @@ def generate_data_type(schema: Mapping[str, Any]) -> str:
         if "nullable" in schema and schema["nullable"]:
             data_type = data_type + " | null"
     return data_type
+
+
+def render_object_details(schema: Mapping[str, Any], spacing: int) -> str:
+    ## TODO: figure out how to render the properties into properly formatted html / string
+    object_details = "<ul>\n"
+    if "items" in schema:
+        object_details = render_object_details(schema["items"], spacing + 4)
+    else:
+        object_values = schema["properties"]
+        for value in object_values:
+            description = object_values[value]["description"]
+            data_type = generate_data_type(object_values[value])
+            object_details += render_description(description, spacing, data_type, value)
+        object_details += "</ul>\n"
+    return object_details
+
+
+def render_description(description: str, spacing: int, data_type: str, value: str) -> str:
+    description = description.replace("\n", " ")
+    return (
+        (spacing * " ")
+        + "<li><code>"
+        + value
+        + "</code>: "
+        + '<span class="api-field-type">'
+        + data_type
+        + "</span> "
+        + description
+        + "</li>\n"
+    )
```

Comment:
Constructing HTML with string manipulation is error-prone, as demonstrated in the example above; I'd consider using some of the `Element` and `SubElement` classes in `zerver/lib/markdown/__init__.py` (and the upstream markdown library) for constructing HTML in the markdown processor.

Also, it can introduce security bugs were these strings user-controlled, because (for example) the content of the code block isn't being automatically HTML-escaped.